core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,47 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+
+                        // NOTE: the edge is only defined for the `u` collection and the `globus_coll` collection.
+                        //  The getUserFromClientID method shows that the user account can also be stored in a UUID collection
+                        //  and in the future support domain accounts. For now, we must ensure that the edge is u to globus_coll
+                        const user_doc = g_db.u.exists(user_id);    // TODO: user doc only necessary for edge key, user_doc._key must not have invalid character '/'
+                                                                    //  which should be kicked out by arango
+                        if (!user_doc) {
+                            throw [g_lib.ERR_INVALID_PARAM, "Error writing to globus_token. No such user in u collection '" + user_id + "'"];
+                        }
+
+                        // find or insert collection
+                        const collection_search_key = other_token_data;
+                        let globus_collection = g_db.globus_coll.exists({_key: collection_search_key});
+                        if (!globus_collection) {
+                            // TODO: what other information should live on the globus_coll document?
+                            globus_collection = g_db.globus_coll.save({_key: collection_search_key, name: "some name we could look up"});
+                        }
+
+                        const token_key = user_doc._key + "_" + globus_collection._key + "_" + token_type;    // TODO: key formatting/convention
```

Comment:
What should we do for these token keys? I went with the mapping of u:globus_coll:type because I feel like that would make the most sense but I am likely not seeing everything.

Reply:
Here's my suggestion:
`globus_coll:type:u`


1. Put the `globus_collection.key` first because:
  * It's the primary scope/context for the token
  * Makes it easier to query all tokens for a specific collection
  * Aligns with how Globus structures their scopes (collection UUID first)
2. Put `token_type` second because:
  * Different token types may be needed for the same collection
  * Helps organize tokens by their purpose/scope
  * Makes it easy to query all tokens of a specific type for a collection                                                                                                                                                              
3. Put `user_doc.key` last because:
  * Multiple users may have tokens for the same collection/type combination
  * Follows principle of most general to most specific
  * Makes hierarchical organization clearer                                                                                                                                                                                            

This structure would make it easier to:

* Query all tokens for a collection
* Query all tokens of a specific type for a collection
* Query a specific user's token for a collection/type combination

The format would look like: `1cbaaee5-b938-4a4e-87a8-f1ec4d5d92f9:1:user123`

Reply:
Ok! I like this suggestion. One of the main reasons I had user first was that the user was context we likely always have, but in a sense we can also use that fact to reduce options earlier in the search, and also moving from more general to more specific makes a lot of sense.

Thank you for the input, I'll get this order changed 🤠